src/tokenizer/lower_caser.rs

```diff
@@ -1,42 +1,24 @@
 use std::mem;
 
-use super::{Token, TokenFilter, TokenStream, Tokenizer};
+use super::{Token, TokenFilter, TokenStream};
 
 /// Token filter that lowercase terms.
 #[derive(Clone)]
 pub struct LowerCaser;
 
 impl TokenFilter for LowerCaser {
-    type Tokenizer<T: Tokenizer> = LowerCaserFilter<T>;
+    type OutputTokenStream<T: TokenStream> = LowerCaserTokenStream<T>;
 
-    fn transform<T: Tokenizer>(self, tokenizer: T) -> Self::Tokenizer<T> {
-        LowerCaserFilter {
-            tokenizer,
+    fn filter<T: TokenStream>(&self, token_stream: T) -> Self::OutputTokenStream<T> {
+        LowerCaserTokenStream {
+            tail: token_stream,
             buffer: String::new(),
```

Comment:
you are reverting here the work done by Pascal to reduce the number of allocation we have. Is this necessary?

Reply:
@Pseitz can you confirm?

Reply:
Yes indeed :/

On each `tokenizer.token_stream`, it will call `filter` and thus will do a `String::new()`.

That was not the case before.

Reply:
I'd prefer if we can have an instance with a buffer in the pipeline like before

Reply:
Yes, of course. I should have fixed by adding the allocation at the `TokenFilter` level.